app/controllers/newflow/login_signup_controller.rb

```diff
@@ -327,15 +327,13 @@ def change_password
     end
 
     def exit_accounts
-      referrer_params = extract_params(request.referrer)
-
-      if (redirect_param = referrer_params[:r])
+      if (redirect_param = extract_params(request.referrer)[:r])
         if Host.trusted?(redirect_param)
           redirect_to(redirect_param)
         else
           raise Lev::SecurityTransgression
         end
-      elsif (redirect_uri = referrer_params[:redirect_uri])
+      elsif !signed_in? && (redirect_uri = extract_params(stored_url)[:redirect_uri])
```

Comment:
consider use of safe assignment inside a condition https://github.com/rubocop-hq/ruby-style-guide#safe-assignment-in-condition

Reply:
Yeap, I'm doing that. It says "[...] unless the assignment is wrapped in parentheses"

Reply:
yeah, ok.  i'm of the opinion it's best to avoid all assignments in IFs, because i know for a fact that they are a source of bugs down the road.  esp deeper nested IFs and scoping.  just my 2 cents....